extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.js

```diff
@@ -0,0 +1,437 @@
+require(
+	['jquery', 'wikia.window', 'wikia.loader', 'wikia.mustache', 'wikia.log'],
+	function ($, window, loader, mustache, log) {
+		'use strict';
+
+		const notificationTypes = {
+			discussionUpvotePost: 'discussion-upvote-post',
+			discussionUpvoteReply: 'discussion-upvote-reply',
+			discussionReply: 'discussion-reply',
+			announcement: 'announcement'
+		}, logTag = 'on-site-notifications';
+
+		/**
+		 * Gets timestamp from ISO string date
+		 * @param {string} date
+		 * @returns {number} - timestamp
+		 */
+		function convertToTimestamp(date) {
+			return (new Date(date)).getTime() / 1000;
+		}
+
+		/**
+		 * Gets ISO string from timestamp
+		 * @param {string} timestamp
+		 * @returns {string} - ISO string
+		 */
+		function convertToIsoString(timestamp) {
+			return new Date(timestamp * 1000).toISOString();
+		}
+
+		function getSafely(obj, path) {
+			return path.split(".").reduce(function (acc, key) {
+				return (typeof acc == "undefined" || acc === null) ? acc : acc[key];
+			}, obj);
+		}
+
+		function TextFormatter() {
+			function bold(text) {
+				return text ? '<b>' + text + '</b>' : text;
+			}
+
+			function fillArgs(message, args) {
+				return Object.keys(args).reduce(function (acc, key) {
+					return acc.replace('__' + key + '__', args[key])
+				}, message);
+			}
+
+			this.getText = function (notification) {
+				if (notification.type === notificationTypes.discussionReply) {
+					return this._getReplyText(notification);
+				} else if (notification.type === notificationTypes.discussionUpvotePost) {
+					return this._getPostUpvoteText(notification)
+				} else if (notification.type === notificationTypes.discussionUpvoteReply) {
+					return this._getReplyUpvoteText(notification);
+				} else {
+					return notification.title;
+				}
+			};
+
+			this._getReplyText = function (notification) {
+				const key = this._getReplyKey(notification.title, notification.totalUniqueActors);
+				const message = window.mw.message(key).parse();
+				var args = {
+					postTitle: bold(notification.title)
+				};
+
+				if (notification.totalUniqueActors > 2) {
+					args.mostRecentUser = notification.latestActors[0].name;
+					args.number = notification.totalUniqueActors - 1;
+				} else if (notification.totalUniqueActors == 2) {
+					args.firstUser = notification.latestActors[0].name;
+					args.secondUser = notification.latestActors[1].name;
+				} else {
+					args.user = notification.latestActors[0].name;
+				}
+
+				return fillArgs(message, args);
+			};
+
+			this._getReplyKey = function (title, totalUniqueActors) {
+				const user = totalUniqueActors <= 1 ? '' :
+					totalUniqueActors == 2 ? 'two-users-' : 'multiple-users-';
+				return 'notifications-replied-by-' + user + (title ? 'with-title' : 'no-title');
```

Comment:
I know it's hard to change that now but in MW you could use special message syntax to handle those cases: https://www.mediawiki.org/wiki/Localisation#Message_parameters